### ISSUE TITLE
Valid line for pointer

### ISSUE THREAD

**Issue by samixchoumi** [Riverside-Software/sonar-openedge#266]
I got this error after asking the scan to tell me where it find ASSIGN, but my procedure have only 709 lines anf after looking at the scan some time it show me that i've used ASSIGN at some lines but ASSIGN isn't here. In the procedure i've only 14 ASSIGN but the scan show me 73 ASSIGN.
Below it's the message :
```
[sonar:sonar] 14:13:32.305 ERROR - Runtime exception was caught '730 is not a valid line for pointer. File [moduleKey=acme-test:jenkins-BUILD46, relative=lmcaldis.p, basedir=C:\User\workspace\BUILD46\temp\src] has 709 line(s)' - Please report this issue : 
[sonar:sonar] 14:13:32.305 ERROR -   org.sonar.api.internal.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
[sonar:sonar] 14:13:32.305 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.checkValid(DefaultInputFile.java:249)
[sonar:sonar] 14:13:32.305 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.newPointer(DefaultInputFile.java:186)
[sonar:sonar] 14:13:32.305 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.selectLine(DefaultInputFile.java:209)
[sonar:sonar] 14:13:32.305 ERROR -   org.sonar.plugins.openedge.api.checks.OpenEdgeCheck.reportIssue(OpenEdgeCheck.java:68)
```

**Comment by gquerret** [comment]
Which rule reports this issue ? If that's an XREF rule, then the culprit is OpenEdge as line numbers are retrieved from the XML XREF. If not an XREF rule, there are still some cases where a combination of include files and preprocessor can cause this kind of problem ; that would need however some more information.

**Comment by samixchoumi** [comment]
I think that's the same error as my other post but this time the test file hasn't any line to report the issue. And the rule isn't an XREF rule, it's a proparse check

**Comment by gquerret** [comment]
So is that a duplicate of #267 ?

**Comment by samixchoumi** [comment]
No, because i've updated everything to 2.0.1 and i still got this error.

```
[sonar:sonar] 15:26:03.596 ERROR - Runtime exception was caught '220 is not a valid line for pointer. File [moduleKey=acme-test:jenkins-BUILD46.06, relative=blope.p, basedir=C:\User\workspace\BUILD46.06\temp\src] has 133 line(s)' - Please report this issue : 
[sonar:sonar] 15:26:03.597 ERROR -   org.sonar.api.internal.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
[sonar:sonar] 15:26:03.597 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.checkValid(DefaultInputFile.java:249)
[sonar:sonar] 15:26:03.597 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.newPointer(DefaultInputFile.java:186)
[sonar:sonar] 15:26:03.597 ERROR -   org.sonar.api.batch.fs.internal.DefaultInputFile.selectLine(DefaultInputFile.java:209)
[sonar:sonar] 15:26:03.597 ERROR -   org.sonar.plugins.openedge.api.checks.OpenEdgeCheck.reportIssue(OpenEdgeCheck.java:68)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitNode(testRule.java:225)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:60)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.597 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.598 ERROR -   com.acme.rules.testRule.visitTree(testRule.java:66)
[sonar:sonar] 15:26:03.598 ERROR -   com.acme.rules.testRule.execute(testRule.java:57)
[sonar:sonar] 15:26:03.598 ERROR -   com.acme.rules.testRule.execute(testRule.java:19)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:204)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:136)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
[sonar:sonar] 15:26:03.599 ERROR -   org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.batch.bootstrapper.Batch.execute(Batch.java:77)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:48)
[sonar:sonar] 15:26:03.613 ERROR -   sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[sonar:sonar] 15:26:03.613 ERROR -   sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[sonar:sonar] 15:26:03.613 ERROR -   sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 15:26:03.613 ERROR -   java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
[sonar:sonar] 15:26:03.613 ERROR -   java.security.AccessController.doPrivileged(Native Method)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.runner.api.Runner.execute(Runner.java:100)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.ant.SonarTask.launchAnalysis(SonarTask.java:53)
[sonar:sonar] 15:26:03.613 ERROR -   org.sonar.ant.SonarTask.execute(SonarTask.java:48)
[sonar:sonar] 15:26:03.613 ERROR -   org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
[sonar:sonar] 15:26:03.613 ERROR -   sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
[sonar:sonar] 15:26:03.613 ERROR -   sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 15:26:03.613 ERROR -   java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Target.execute(Target.java:390)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Target.performTasks(Target.java:411)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Project.executeTargets(Project.java:1251)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:105)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
[sonar:sonar] 15:26:03.614 ERROR -   sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
[sonar:sonar] 15:26:03.614 ERROR -   sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 15:26:03.614 ERROR -   java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Target.execute(Target.java:390)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Target.performTasks(Target.java:411)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Project.executeTarget(Project.java:1368)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Project.executeTargets(Project.java:1251)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Main.runBuild(Main.java:809)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.Main.startAnt(Main.java:217)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
[sonar:sonar] 15:26:03.614 ERROR -   org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

This rule is just used to find specific statement (like the exemple you gave) and it never use line, except for report issue (node.getline()).

**Comment by gquerret** [comment]
As in #267, please include code sample to reproduce the problem

**Comment by gquerret** [comment]
Closed as not receiving news.
